src/Enable.Extensions.Queuing.RabbitMQ/Internal/RabbitMQConnectionManager.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Collections.Generic;
+using RabbitMQ.Client;
+
+namespace Enable.Extensions.Queuing.RabbitMQ.Internal
+{
+    internal class RabbitMQConnectionManager
+    {
+        private static readonly Dictionary<int, RabbitMQManagedConnection>
```

Comment:
We've used a lot of `lock`s in this class. These can all be replaced if we use `ConcurrentDictionary`, instead of a `Dictionary`.

Reply:
Locking is used to ensure that the entire retrieval and construction lifecycle is not duplicated across threads.

Reply:
Agreed, but manually maintaining locks leads to a lot more code to maintain compared to using one of the thread-safe collections, like `ConcurrentDictionary`. Replacing our use of `Dictionary` with `ConcurrentDictionary` should allow us to remove a load of locks, and simplify the code in this class a bit.